lib/resources/docker.rb

```diff
@@ -0,0 +1,341 @@
+        .add(:mounts,         field: 'mounts')
+        .add(:names,          field: 'names')
+        .add(:networks,       field: 'networks')
+        .add(:ports,          field: 'ports')
+        .add(:running_for,    field: 'runningfor')
+        .add(:sizes,          field: 'size')
+        .add(:status,         field: 'status')
+        .add(:exists?) { |x| !x.entries.empty? }
+  filter.connect(self, :containers)
+
+  attr_reader(:containers)
+  def initialize(containers)
+    @containers = containers
+  end
+end
+
+class DockerImageFilter
+  filter = FilterTable.create
+  filter.add_accessor(:where)
+        .add_accessor(:entries)
+        .add(:ids,              field: 'id')
+        .add(:repositories,     field: 'repository')
+        .add(:tags,             field: 'tag')
+        .add(:sizes,            field: 'size')
+        .add(:digests,          field: 'digest')
+        .add(:created,          field: 'createdat')
+        .add(:created_since,    field: 'createdsize')
+        .add(:exists?) { |x| !x.entries.empty? }
+  filter.connect(self, :images)
+
+  attr_reader(:images)
+  def initialize(images)
+    @images = images
+  end
+end
+
+# For compatability with Serverspec we also offer the following resouses:
+# - docker_container
+# - docker_image
+module Inspec::Resources
+  # This resource helps to parse information from the docker host
+  class Docker < Inspec.resource(1)
+    name 'docker'
+
+    desc "
+      A resource to retrieve information about docker
+    "
+
+    example "
+      describe docker.version do
+        its('Client.Version') { should cmp >= '1.12'}
+      end
+
+      describe docker.inspect('e0825339329b') do
+        its(%w(HostConfig Privileged)) { should eq false }
+        its(%w(HostConfig Privileged)) { should_not eq true }
+      end
+
+      docker.containers.ids.each do |id|
+        describe docker.inspect(id) do
+          its(%w(HostConfig Privileged)) { should eq false }
+          its(%w(HostConfig Privileged)) { should_not eq true }
+        end
+      end
+
+      describe file(docker.path) do
+        it { should exist }
+        it { should be_file }
+        it { should be_owned_by 'root' }
+        it { should be_grouped_into 'root' }
+      end
+
+      describe file(docker.socket) do
+        it { should exist }
+        it { should be_file }
+        it { should be_owned_by 'root' }
+        it { should be_grouped_into 'root' }
+      end
+    "
+
+    def containers
+      DockerContainerFilter.new(parse_containers)
+    end
+
+    def images
+      DockerImageFilter.new(parse_images)
+    end
+
+    # returns the path of the docker service
+    # only works for Debian 8+, Ubuntu 16.04+ and CentOS 7+ where systemd is installed
+    def path
+      # TODO: use exception, once https://github.com/chef/inspec/issues/1205 is fixed
+      return nil unless is_systemd_system
+
+      cmd = inspec.command('systemctl show -p FragmentPath docker.service')
```

Comment:
Since we're doing something similar both here, and in `#socket`, maybe we should extract this to a `#systemd_value` helper method:

```
def systemd_value(unit_name, key)
  # return nil if not a systemd system
  # do the inspec.command call to systemctl, passing in #{unit_name}
  # return nil if nonzero
  # parse the stuff
  # return parsed_stuff[key]
end
```

That would simplify both of these to:

```
def path
   systemd_value('docker.service', 'FragmentPath')
end

def socket
  systemd_value('docker.socket', 'FragmentPath')
end
```

... and makes it easier to test.